### PR TITLE
Allow toolchains to bet set

### LINKED ISSUE #114: Toolchain configuration
Is there a reason why the `TOOLCHAIN_*` variables are set with the `override` prefix in `Configuration.mk`?

The values for `TOOLCHAIN_rv32im[a]c` are currently `riscv64-unknown-elf`.
Some packages (e.g. Arch, which is also referenced in the README.md) have binaries which have a prefix `riscv64-elf`.
It should also be possible to compile with `riscv32-unknown-elf-gcc`.
Not using `override` would help to configure which compiler is used locally.

The configuration dump of `Configuration.mk` uses the variable `TOOLCHAIN` but as far as I can see it is never used, to this output is a bit confusing.

### PR DESCRIPTION
**Pull request by towoe**
Remove keyword `override` so the variables are changeable with make
arguments.

Fixes #114

### CODE REVIEW

File: Configuration.mk [hunk 1]
```diff
@@ -80,11 +80,11 @@ ELF2TAB_ARGS += -n $(PACKAGE_NAME)
...
-override TOOLCHAIN_cortex-m0 = arm-none-eabi
-override TOOLCHAIN_cortex-m3 = arm-none-eabi
-override TOOLCHAIN_cortex-m4 = arm-none-eabi
-override TOOLCHAIN_rv32imac = riscv64-unknown-elf
-override TOOLCHAIN_rv32imc = riscv64-unknown-elf
+TOOLCHAIN_cortex-m0 = arm-none-eabi
+TOOLCHAIN_cortex-m3 = arm-none-eabi
+TOOLCHAIN_cortex-m4 = arm-none-eabi
+TOOLCHAIN_rv32imac = riscv64-unknown-elf
+TOOLCHAIN_rv32imc = riscv64-unknown-elf
```
Comment (ppannuto) [inline Configuration.mk:87]:
```suggestion
TOOLCHAIN_cortex-m0 := arm-none-eabi
TOOLCHAIN_cortex-m3 := arm-none-eabi
TOOLCHAIN_cortex-m4 := arm-none-eabi
TOOLCHAIN_rv32imac := riscv64-unknown-elf
TOOLCHAIN_rv32imc := riscv64-unknown-elf
```

While we're fixing things, we generally try avoid lazy binding in our makefiles... not really sure what happened to TOOLCHAIN :/

**Review by ppannuto — CHANGES_REQUESTED** [review]
image
Well this is about to be a silly cycle

**Review by ppannuto — APPROVED** [review]
bors r+

**Comment by towoe** [comment]
Thanks for the follow up commit @ppannuto

**Pull request merged**